### DISCUSSION TITLE
Add data SC in Pilot Pattern

### DISCUSSION THREAD

**Discussion by Mio-TW** [NVlabs/sionna#410] [Q&A] [answered]
Dear Sionna Team,

I see the example [here](https://nvlabs.github.io/sionna/api/ofdm.html#pilotpattern) for pilot pattern construction.
There are two streams, and both of them have "mask" components reserved for others transmitter.
Now if I have only one stream but I want the frequency domain to transmit more data.
How could I design the resource grid?
For example, using SC index 0 ,2, 4, 6, ... for pilot and SC index 1, 3, 5, 7, ... for data?
And this _resource_grid_ need to directly be called by [LS channel estimator](https://nvlabs.github.io/sionna/api/ofdm.html#lschannelestimator).

Thank you so much.

**Comment by jhoydis** [comment]
Hi @Mio-TW,

You can freely design pilot patterns. The option you propose should work. 
The `LSCHannelEstimator` needs this resource grid as input parameter during initialization.

**Reply by Mio-TW** [reply]
@jhoydis thank you so much.
Could you please advise on how to design a pattern like this? 
Do you have any example code?
Thanks!

**Reply by SebastianCa** [reply, marked as answer]
Hi @Mio-TW,

You can find an example of custom `PilotPattern` in the [API documentation](https://nvlabs.github.io/sionna/api/ofdm.html#pilot-pattern). In the example, we use QPSK pilots, but you can assign any other pattern as well. Keep in mind that typically there should be a power normalization of the pilots involved (or use a carefully adjusted pilot boosting as done in 5G NR).

Furthermore, the [MIMO OFDM notebook](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html#OFDM-Resource-Grid-&-Pilot-Pattern) provides an example of how to initialize the `ResourceGrid`. Instead of `pilot_pattern="kronecker"`, you would need to provide your custom `PilotPattern`.

I hope this helps.
